lxd/images.go

```diff
@@ -107,30 +151,22 @@ func imagesPost(d *Daemon, r *http.Request) Response {
 	}
 
 	f, err := ioutil.TempFile(dirname, "image_")
+	defer f.Close()
+
 	if err != nil {
 		return InternalError(err)
 	}
 
 	fname := f.Name()
 
-	_, err = io.Copy(f, r.Body)
+	reqFileReader := NewHttpRequestFileReader(r.Body)
+	size, err := io.Copy(f, reqFileReader)
```

Comment:
The use of NewHttpRequestFileReader seems a bit awkward here; can we not just do an io.Copy(sha256writer, r.Body)?